### PR TITLE
Add FTPS Transform

### PR DESCRIPTION
**Pull request by aaspring**
With correct file this time.

### CODE REVIEW

**Comment by zakird** [comment]
Let's subclass instead of calling a random method on a different protocol.

File: ztag/transforms/ftp.py [hunk 1]
```diff
@@ -31,3 +32,40 @@ def _transform_object(self, obj):
...
+
+    def __init__(self, *args, **kwargs):
+        super(FTPSTransform, self).__init__(*args, **kwargs)
+
+    def _transform_object(self, obj):
+        ftp_banner = obj
+        ftp = Transformable(obj)
+        zout = ZMapTransformOutput()
+        error = ftp['error'].resolve()
+        if error is not None:
```
Comment (zakird) [inline ztag/transforms/ftp.py]: can be

is error:

Comment (ppannuto) [inline ztag/transforms/ftp.py]: My reading of
http://stackoverflow.com/questions/7816363/if-a-vs-if-a-is-not-none and the
linked part of PEP8 would suggest leaving it as a comparison to None. I
think it's clearer that way as well.

File: ztag/transforms/ftp.py [hunk 2]
```diff
@@ -17,12 +18,12 @@ def _transform_object(self, obj):
         ftp = Transformable(obj)
         zout = ZMapTransformOutput()
         error = ftp['error'].resolve()
-        if error is not None:
+        if error != None:
```
Comment (dadrian) [inline ztag/transforms/ftp.py]: Why did these get changed? Also, per PEP8, comparisons to singletons (such as `None`), should always be done with the `is` operator. https://www.python.org/dev/peps/pep-0008/#programming-recommendations

File: ztag/transforms/ftp.py [hunk 3]
```diff
@@ -31,3 +32,40 @@ def _transform_object(self, obj):
...
+
+    def __init__(self, *args, **kwargs):
+        super(FTPSTransform, self).__init__(*args, **kwargs)
+
+    def _transform_object(self, obj):
+        ftp_banner = obj
+        ftp = Transformable(obj)
+        zout = ZMapTransformOutput()
+        error = ftp['error'].resolve()
+        if error != None:
```
Comment (dadrian) [inline ztag/transforms/ftp.py]: If banner is non-null, but we timeout/error/are told there's no support for AuthTLS and AuthSSL, will the error field in the ZGrab output be non-null?

**Comment by zakird** [comment]
@aaspring what's the status of this pull request? It has noted problems. Do you plan to fix or should we close it?

**Pull request closed without merging**